contracts/JumpBall.cdc

```diff
@@ -0,0 +1,379 @@
+/*
+    The JumpBall contract facilitates competitive games by securely managing players' NFTs.
+    It creates game-specific instances to isolate gameplay, store NFTs, and enforce game rules.
+    NFTs are released to the winner or returned to their original owners based on the game's outcome.
+
+    Authors:
+        Corey Humeston: corey.humeston@dapperlabs.com
+*/
+
+import NonFungibleToken from 0xf8d6e0586b0a20c7
+
+access(all) contract JumpBall {
+    // Events
+    access(all) event GameCreated(gameID: String, creator: Address, startTime: UFix64)
+    access(all) event OpponentAdded(gameID: String, opponent: Address)
+    access(all) event NFTDeposited(gameID: String, nftID: UInt64, owner: Address)
+    access(all) event NFTReturned(gameID: String, nftID: UInt64, owner: Address)
+    access(all) event NFTAwarded(gameID: String, nftID: UInt64, previousOwner: Address, winner: Address)
+    access(all) event WinnerDetermined(gameID: String, winner: Address)
+    access(all) event TimeoutClaimed(gameID: String, claimant: Address)
+
+    // Player resource for game participation
+    access(all) resource Player {
+        access(all) let address: Address
+        access(all) let collectionCap: Capability<&{NonFungibleToken.Collection}>
+        access(all) var metadata: {String: AnyStruct}
+
+        init(address: Address, collectionCap: Capability<&{NonFungibleToken.Collection}>) {
+            self.address = address
+            self.collectionCap = collectionCap
+            self.metadata = {}
+        }
+
+        // Helper function for player metadata
+        access(all) fun setMetadata(key: String, value: AnyStruct) {
+            self.metadata[key] = value
+        }
+
+        access(all) fun getMetadata(key: String): AnyStruct? {
+            return self.metadata[key]
+        }
+
+        // Create a new game
+        access(all) fun createGame(gameID: String, startTime: UFix64, gameDuration: UFix64, selectedStatistic: String): String {
```

Comment:
We should add an entitlement for `createGame` and `addOpponent` to ensure only the owner/entitled reference to the Player resource can call the methods in case a capability for the resource is published